contracts/masset/BasketManager.sol

```diff
@@ -485,6 +503,68 @@ contract BasketManager is
         emit BassetRemoved(bAsset.addr);
     }
 
+    /**
+     * @dev Transfers all collateral from one lending market to another - used initially
+     *      to handle the migration between Aave V1 and Aave V2. Note - only supports non
+     *      tx fee enabled assets
+     * @param _bAssets Array of basket assets to migrate
+     * @param _newIntegration Address of the new platform integration
+     */
+    function migrateBassets(
+        address[] calldata _bAssets,
+        address _newIntegration
+    )
+        external
+        onlyGovernor
+    {
+        uint256 len = _bAssets.length;
+        require(len > 0, "Must migrate some bAssets");
+
+        for(uint i = 0; i < len; i++){
+            // 1. Check that the bAsset is in the basket
+            address bAsset = _bAssets[i];
+            (bool inBasket, uint8 index) = _isAssetInBasket(bAsset);
+            require(inBasket, "bAsset does not exist");
+
+            // 2. Withdraw everything from the old platform integration
+            IPlatformIntegration oldIntegration = IPlatformIntegration(integrations[index]);
+            require(address(oldIntegration) != _newIntegration, "Must transfer to new integrator");
+            // 2.1. Withdraw from the lending market
+            uint256 lendingBal = oldIntegration.checkBalance(bAsset);
+            uint256 cache = IERC20(bAsset).balanceOf(address(oldIntegration));
+            oldIntegration.withdraw(address(this), bAsset, lendingBal, false);
+            // 2.2. Withdraw from the cache, if any
+            oldIntegration.withdrawRaw(address(this), bAsset, cache);
+            uint256 total = lendingBal.add(cache);
+
+            // 3. Update the integration address for this bAsset
+            integrations[index] = _newIntegration;
+
+            // 4. Deposit everything into the new
+            //    This should fail if we did not receive the full amount from the platform withdrawal
+            // 4.1. Deposit all bAsset
+            IERC20(bAsset).safeTransfer(_newIntegration, total);
+            IPlatformIntegration newIntegration = IPlatformIntegration(_newIntegration);
+            newIntegration.deposit(bAsset, lendingBal, false);
+            // 4.2. Check balances
+            uint256 newLendingBal = newIntegration.checkBalance(bAsset);
+            uint256 newCache = IERC20(bAsset).balanceOf(address(newIntegration));
+            uint256 upperMargin = 1001e15;
```

Comment:
What's the reason for enforcing an upper bound?

Reply:
Incase for some reason the new integrations `checkBalance` call returns some number with completely different precision.. it makes sense to just check here in case we cause hyper inflation.